src/test/groovy/unit/org/pih/warehouse/inventory/TransactionTypeSpec.groovy

```diff
@@ -0,0 +1,91 @@
+package org.pih.warehouse.inventory
+
+import spock.lang.Unroll
+
+import grails.testing.gorm.DomainUnitTest
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class TransactionTypeSpec  extends Specification implements DomainUnitTest<TransactionType> {
+
+
+    @Test
+    @Unroll('TransactionType.validate() with name: #value should return #expected with errorCode: #expectedErrorCode')
+    void 'name field validation'() {
```

Comment:
@EWaterman I made some modifications to the tests and this is what the output looks like in the console. 

![image](https://github.com/openboxes/openboxes/assets/1162935/d70fd719-8cf0-4123-b014-e3985fb82d8e)

and the test report 

![image](https://github.com/openboxes/openboxes/assets/1162935/21334b98-bec0-4fdf-9c84-6ccbcbe1ca2a)

If we don't want data (especially long names) in the test spec description, then we could probably break the tests into "test with valid data" vs "tests against invalid data"

So, instead of this 
```
org.pih.warehouse.inventory.TransactionTypeSpec > Should return false with error code nullable when validating transaction type given name null PASSED

org.pih.warehouse.inventory.TransactionTypeSpec > Should return false with error code maxSize.exceeded when validating transaction type given name aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa PASSED

org.pih.warehouse.inventory.TransactionTypeSpec > Should return true with error code null when validating transaction type given name bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb PASSED
```
we'd see this
```
org.pih.warehouse.inventory.TransactionTypeSpec > Should return false with error code nullable when validating transaction type given an invalid name PASSED

org.pih.warehouse.inventory.TransactionTypeSpec > Should return false with error code maxSize.exceeded when validating transaction type given an invalid name PASSED

org.pih.warehouse.inventory.TransactionTypeSpec > Should return true with error code null when validating transaction type given a valid name PASSED
```
Or we could add a test description to the table 
```
adjective    || value        || expected | expectedErrorCode
null         || null         || false    | 'nullable'
really long  || 'a' * 256    || false    | 'maxSize.exceeded'
longish      || 'b' * 255    || true     | null
valid        || 'Dummy Name' || true     | null
```
So the test spec
```
def 'Should return #expected with error code #expectedErrorCode when validating transaction type given a #adjective name'() {
```
would resolve to 
```
org.pih.warehouse.inventory.TransactionTypeSpec > Should return false with error code maxSize.exceeded when validating transaction type given a really long PASSED
```

Reply:
yeah I agree in that I don't think we necessarily need to be ultra-verbose in specifying the exact inputs in the test name, but also I'm not sure if I'm a fan of needing to specify additional adjective fields just to populate the test name properly. I'm not sure it's the hugest deal either way. Ultimately I feel like as long as we stick to a general "X should return/do Y when Z" format, we're good.